docs/decisions/0003-static-builds-to-client-side-react.rst

```diff
@@ -0,0 +1,55 @@
+
+Despite these primary benefits, using Gatsby comes with a few tradeoffs that influence whether it is the right choice for the Enterprise learner portal moving forward. At the time Gatsby was introduced, the Enterprise learner portal was essentially a single dashboard page for enterprise learners to see their current, upcoming, and completed course enrollments associated with their Enterprise. This page also contains basic branding specific to the Enterprise (i.e., logo, colors, etc.). However, as we build out a more robust learner experience within the Enterprise learner portal, we plan to add additional pages such as a search/filter course discovery experience and course detail pages. Each of these pages will persist the Enterprise Customer's logo, name, and branding.
+
+Taking the course detail pages as an example, let's assume a worst-case scenario where we have 500 Enterprise Customers, each with a course catalog that grants access to all edX content (i.e., ~2500 courses). In this scenario, because each course detail page has branding that is unique to each individual Enterprise Customer, we would potentially need to build 1,250,000 static pages to support the course catalogs of those 500 Enterprise Customers. This would lead to long build times that would slow down development, testing, and rolling out production bug fixes.
+
+An argument might be made that we could implement a hybrid approach, where we only build each course page once and dynamically fetch the Enterprise Customer's data (e.g., logo, name, branding) client-side. However, at that point, we would not really be taking full advantage of Gatsby as a static site generator. Resources online point to scenarios such as this, where most of an application uses dynamic data, as rationale for not using Gatsby (see References).
+
+********
+Decision
+********
+
+While it would be great if the Enterprise learner portal could take advantage of the benefits for using Gatsby (e.g., faster page loads), it does not make as much sense given the tradeoffs mentioned above. This is because the Enterprise learner portal relies mostly on dynamic data in such a way that would not allow us to realize the advantage of a statically rendered app. Additionally, the Enterprise learner portal does not have a requirement for supporting SEO best practices since it is used by existing customers (presumably with direct access to the URL) as opposed to a marketing site that will be crawled by search engine bots.
+
+That said, we will be migrating the Enterprise learner portal off of Gatsby and its multi-site build process / infrastructure. Instead, the Enterprise learner portal will become a standard client-side React application (microfrontend) that utilizes many of the recent advancements in the frontend ecosystem at edX (e.g., ``@edx/frontend-build``, ``@edx/frontend-platform``).
+
```

Comment:
Another reason I would add was that we didn't know how we would scale deployment of the enterprise learner portal and easily make it available for more enterprises. Because each portal was statically built, we would need to build a mechanism for changes to the page by an internal admin (aka ECS team) to trigger a rebuild and deploy without intervention from Engineering. This is ok for masters because there are less than 20 "portals", but potential/eventual scale for enterprise is 500+. Removing the static build step and dynamically fetching the data means these changes will happen instantaneously without engineering intervention.